server/src/main/java/org/apache/iotdb/db/qp/executor/QueryProcessExecutor.java

```diff
@@ -244,18 +271,47 @@ public boolean insert(InsertPlan insertPlan) throws ProcessorException {
     try {
       String[] measurementList = batchInsertPlan.getMeasurements();
       String deviceId = batchInsertPlan.getDeviceId();
-      MNode node = mManager.getNodeByDeviceIdFromCache(deviceId);
+      Object[] values = collectFirstElements(batchInsertPlan.getColumns());
+      IoTDBConfig conf = IoTDBDescriptor.getInstance().getConfig();
+
+      try {
+        mManager.checkDeviceId(deviceId);
```

Comment:
This method is called very frequently and it is not a light operation. I am afraid that if you do not add a cache mechanism for it, the performance will downgrade.